### DISCUSSION TITLE
setValue not trigger onChange revalidation

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [react-hook-form/react-hook-form#8273] [Q&A] [answered]
I have complex hard component which i register in useEffect and use setValue for set data.

When i delete value error field not trigger reValidate

**Comment by bluebill1049** [comment, marked as answer]
`setValue('name', 'value', { shouldValidate: true })` ?

**Reply by nvmnghia** [reply]
@bluebill1049 I bumped into this too and adopt the same solution. Could you please explain why this is not the default?

Also, it is said in [`setValue()` docs' Rule section](https://react-hook-form.com/api/useform/setvalue) that `setValue()` *does* trigger validation in some case, without `shouldValidate` (it doesn't mentioned that, so I infer it that way). However, I can't replicate it, although I already subscribe to `errors`. My [codesandbox](https://codesandbox.io/s/sweet-lumiere-90g6vi?file=/src/App.js). Quickly click Submit, the form validates, error appears. After ~3s, the form fills data, the error is still there.

I assume new render = validation.

**Reply by Upinwv** [reply]
![image](https://user-images.githubusercontent.com/99217560/171932677-1efc9498-e02a-4218-971c-25ceeca72f84.png)
